sound/soc/codecs/rt700-sdw.c

```diff
@@ -418,10 +418,12 @@ static int rt700_interrupt_callback(struct sdw_slave *slave,
 	dev_dbg(&slave->dev,
 		"%s control_port_stat=%x", __func__, status->control_port);
 
-	if (status->control_port & 0x4) {
+	mutex_lock(&rt700->irq_lock);
```

Comment:
I originally proposed a spin-lock because the function is *called* `..._interrupt_callback()`. Having checked that though I see, that it's actually called from a work context, so it should be ok. Still, I'd either add a comment to these functions, that they aren't actually called in interrupt context, or maybe even change their names, or add a comment to the mutex here, or actually still use a spinlock: the locked code sequences are short enough and we don't expect contention too often here... also a mutex, called `irq_lock` gives me a strange feeling in the stomach :-)

Reply:
The 'interrupt callback' is a generic term, it's intended to be used in interrupt thread or workqueue context, ie. not in interrupt context. I don't think it matters if it's a workqueue or interrupt thread, does it?

I can add a comment on the mutex vs. spinlock. The part on disabling the interrupt can be actually quite slow, it will require one command sent on the bus and an interrupt to handle the response, a spinlock isn't appropriate here.

I am not very good at naming. Maybe disable_irq_lock to go along the disable_irq flag?